ir/pattern/mod.rs

```diff
@@ -32,6 +32,9 @@ pub mod expression;
 pub mod function_call;
 pub mod nested_pattern;
 
+#[derive(Debug, Copy, Clone, Hash, Eq, PartialEq)]
+pub struct BranchID(pub u16);
```

Comment:
It's not clear to me from the code or the PR description yet what the relationship is between Provenance (and that 64 limit I saw), and this BranchID? Both of them seem to impose some limit on query structure we can represent, but which one does what?

Reply:
Ah I see where the confusion is from. This could easily be u8 (or technically u6).